core/types/calls.cc

```diff
@@ -2098,6 +2125,7 @@ const vector<Intrinsic> intrinsicMethods{
     {Symbols::T(), Intrinsic::Kind::Singleton, Names::proc(), &T_proc},
 
     {Symbols::T_Generic(), Intrinsic::Kind::Instance, Names::squareBrackets(), &T_Generic_squareBrackets},
+    {Symbols::T_Generic(), Intrinsic::Kind::Instance, Names::attachedClassNew(), &T_Generic_attachedClassNew},
```

Comment:
Can this go on Magic? It's more similar to methods already there

Reply:
It needs to exist on the class that is using it. If it were to go on Magic, it would dispatch to the constructor for Magic.

Reply:
```
Magic.construct_class(YourRenericClass, *constructorArgs)
```
is what we do for all the others

Reply:
so you have your type for what would be `self` in first arg.

Reply:
So would the constructor turn into `T.attached_class_new` in the runtime, then? Would we need to pass the class name through there as well, or is there a way to grab the `self` of the calling context?

Reply:
I've reworked the intrinsic so that it dispatches to the constructor correctly, and added a load of tests. Thanks for pushing back on the special constructor, that wasn't the right solution :)